vars/patchDockerImage.groovy

```diff
@@ -0,0 +1,111 @@
+    build_date = build_time[0..3] + build_time[5..6] + build_time[8..9]
+
+    def build_qualifier = inputManifest.build.qualifier
+
+    if (build_qualifier != null && build_qualifier != 'null') {
+        build_qualifier = "-" + build_qualifier
+    }
+    else {
+        build_qualifier = ''
+    }
+
+    if (artifactUrlX64 == null || artifactUrlARM64 ==  null) {
+    echo 'Skipping docker build, one of x64 or arm64 artifacts was not built.'
+    } else {
+        echo 'Trigger docker-build'
+        dockerBuild: {
+            build job: 'docker-build',
+            parameters: [
+                string(name: 'DOCKER_BUILD_GIT_REPOSITORY', value: 'https://github.com/opensearch-project/opensearch-build'),
+                string(name: 'DOCKER_BUILD_GIT_REPOSITORY_REFERENCE', value: 'main'),
+                string(name: 'DOCKER_BUILD_SCRIPT_WITH_COMMANDS', value: [
+                        'id',
+                        'pwd',
+                        'cd docker/release',
+                        "curl -sSL ${artifactUrlX64} -o ${args.product}-x64.tgz",
+                        "curl -sSL ${artifactUrlARM64} -o ${args.product}-arm64.tgz",
+                        [
+                            'bash',
+                            'build-image-multi-arch.sh',
+                            "-v ${inputManifest.build.version}${build_qualifier}",
+                            "-f ./dockerfiles/${args.product}.al2.dockerfile",
+                            "-p ${args.product}",
+                            "-a 'x64,arm64'",
+                            "-r opensearchstaging/${args.product}",
+                            "-t '${args.product}-x64.tgz,${args.product}-arm64.tgz'",
+                            "-n ${build_number}"
+                        ].join(' ')
+                ].join(' && ')),
+            ]
+        }
+
+        echo 'Trigger docker-copy with tag build date '
+        if (args.rerelease) {
+            dockerCopy: {
+                build job: 'docker-copy',
+                parameters: [
+                    string(name: 'SOURCE_IMAGE_REGISTRY', value: 'opensearchstaging'),
+                    string(name: 'SOURCE_IMAGE', value: "${args.product}:${inputManifest.build.version}${build_qualifier}"),
+                    string(name: 'DESTINATION_IMAGE_REGISTRY', value: 'opensearchstaging'),
+                    string(name: 'DESTINATION_IMAGE', value: "${args.product}:${inputManifest.build.version}${build_qualifier}.${build_number}.${build_date}")
+                ]
+            }
+        }
+
+        echo "Trigger docker-scan for ${args.product} version ${inputManifest.build.version}${build_qualifier}"
+        dockerScan: {
+            build job: 'docker-scan',
+            parameters: [
+                string(name: 'IMAGE_FULL_NAME', value: "opensearchstaging/${args.product}:${inputManifest.build.version}${build_qualifier}")
+            ]
+        }
+
+        echo 'Trigger docker-promote'
+        if(args.rerelease){
```

Comment:
Do we want to do few more tests before pushing to Prod? For example we do not want to push to prod if high CVEs are detected in above docker scan. @bbarani @peterzhuamazon 

Also @Divyaasm please add https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/distribution-build.jenkinsfile#L440 `wait:true` for each job that is being triggered. I think default is `true` by default but better to mention it.

Reply:
Think we have ignored CVE's presence in this implementation promote the image when Docker-Scan and Docker-Validation(enhancement) are successful . The idea is to re-run the Job every week to align with Base OS Image. @bbarani Correct me if I'm wrong.

Reply:
I don't think we can do anything about any CVE at this point in time if it didn't remediated as part of `yum update` as the fix needs to come from base OS team.